app/src/main/java/com/example/evently/data/NotificationDB.java

```diff
@@ -112,6 +114,16 @@ public Promise<List<Notification>> fetchEventNotifications(UUID eventID) {
                 .map(NotificationDB::parseQuerySnapshot);
     }
 
+    /**
+     * Makes a concurrent task for fetching notification by ID.
+     * @param notificationID the id of the notification
+     * @return Reference to the concurrent task yielding to an account (if found).
+     */
+    public Promise<Optional<Notification>> fetchNotification(UUID notificationID) {
+        return promise(notificationsRef.document(notificationID.toString()).get())
+                .map(NotificationDB::parseDocumentSnapshot);
+    }
```

Comment:
Is this used anywhere?

It should return a `PromiseOpt<Notification>`, you can do so by calling `promiseOpt` on the return value. It's okay to leave it as is for now though.